app/assets/javascripts/components/instructor_panel.js.jsx

```diff
@@ -7,6 +7,16 @@ var InstructorPanel = React.createClass({
       action: this.props.queuePop,
     };
   },
+  getQueuePopandPinButtonData: function () {
+    index = 0 
+    if(this.props.getUnpinnedRequests[0] != null)
+    index = this.props.getUnpinnedRequests[0]
+    return {
+      title: 'Queue Pop and Pin',
+      className: this.buttonBaseClass + "huge primary " + (this.props.requests.length <= 0 ? "disabled" : ""),
+      action: (this.props.requests[index] != null ? this.props.pin.bind(null, this.props.requests[index]["id"]) :null),
```

Comment:
Since this component doesn't actually need the full list of unpinned requests, it just needs a method to pin the top request, would you mind moving this logic out to its own function in `course_queue.js.jsx` and passing that in here directly? That way `action` here is just `this.props.pinTopRequest()` or something like that.